File: service/src/main/java/no/nav/melosys/service/dokument/brev/BrevDataA1.java
package no.nav.melosys.service.dokument.brev;

import java.util.Collection;
import java.util.List;

import no.nav.melosys.domain.adresse.StrukturertAdresse;
import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet;
import no.nav.melosys.domain.kodeverk.Land_iso2;
import no.nav.melosys.domain.kodeverk.yrker.Yrkesgrupper;
import no.nav.melosys.domain.person.Persondata;
import no.nav.melosys.service.dokument.brev.mapper.arbeidssted.Arbeidssted;

public class BrevDataA1 extends BrevData {
    public Yrkesgrupper yrkesgruppe;
    public Persondata person;
    public AvklartVirksomhet hovedvirksomhet;

    public List<Arbeidssted> arbeidssteder;
    public Collection<Land_iso2> arbeidsland;
    public boolean erUkjenteEllerAlleEosLand;

    public StrukturertAdresse bostedsadresse;
    public Collection<AvklartVirksomhet> bivirksomheter;
}

File: service/src/main/kotlin/no/nav/melosys/service/dokument/brev/BrevDataA1.kt
package no.nav.melosys.service.dokument.brev

import no.nav.melosys.domain.adresse.StrukturertAdresse
import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet
import no.nav.melosys.domain.kodeverk.Land_iso2
import no.nav.melosys.domain.kodeverk.yrker.Yrkesgrupper
import no.nav.melosys.domain.person.Persondata
import no.nav.melosys.service.dokument.brev.mapper.arbeidssted.Arbeidssted


class BrevDataA1(
    saksbehandler: String? = null,
    fritekst: String? = null,
    begrunnelseKode: String? = null,
    var yrkesgruppe: Yrkesgrupper? = null,
    var person: Persondata? = null,
    var hovedvirksomhet: AvklartVirksomhet? = null,
    var arbeidssteder: List<Arbeidssted>? = null,
    var arbeidsland: Collection<Land_iso2>? = null,
    var erUkjenteEllerAlleEosLand: Boolean = false,
    var bostedsadresse: StrukturertAdresse? = null,
    var bivirksomheter: Collection<AvklartVirksomhet>? = null
) : BrevData(saksbehandler, fritekst, begrunnelseKode)
